File: services/identity-service/src/main/java/br/com/deltaglobalbank/identity/features/signingKeys/rotateSigningKeys/RotateSigningKeyController.java
package br.com.deltaglobalbank.identity.features.signingKeys.rotateSigningKeys;

import br.com.deltaglobalbank.sharedauth.AuthenticatedPrincipal;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.PostMapping;

@Controller
public class RotateSigningKeyController {

    private final RotateSigningKeyUseCase rotateSigningKeyUseCase;

    public RotateSigningKeyController(RotateSigningKeyUseCase rotateSigningKeyUseCase) {
        this.rotateSigningKeyUseCase = rotateSigningKeyUseCase;
    }

    @PostMapping("/admin/signing-keys/rotate")
    @PreAuthorize("hasRole('platform.admin')")
    public ResponseEntity<RotateSigningKeyResponse> rotateSigningKey(
        @AuthenticationPrincipal AuthenticatedPrincipal principal
    ) {
        RotateSigningKeyResponse response = rotateSigningKeyUseCase.execute(principal.subject());
        return ResponseEntity.status(HttpStatus.CREATED).body(response);
    }
}

File: services/identity-service/src/main/kotlin/br/com/deltaglobalbank/identity/features/signingKeys/rotateSigningKeys/RotateSigningKeyController.kt
package br.com.deltaglobalbank.identity.features.signingKeys.rotateSigningKeys

import br.com.deltaglobalbank.sharedauth.AuthenticatedPrincipal
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.security.access.prepost.PreAuthorize
import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.PostMapping

@Controller
class RotateSigningKeyController(
    private val rotateSigningKeyUseCase: RotateSigningKeyUseCase
) {
    @PostMapping("/admin/signing-keys/rotate")
    @PreAuthorize("hasRole('platform.admin')")
    fun rotateSigningKey(
        @AuthenticationPrincipal principal: AuthenticatedPrincipal
    ): ResponseEntity<RotateSigningKeyResponse> {
        val response = rotateSigningKeyUseCase.execute(principal.subject)
        return ResponseEntity.status(HttpStatus.CREATED).body(response)
    }
}